lib/jekyll/commands/new.rb

```diff
@@ -114,6 +115,29 @@ def site_template
         def scaffold_path
           "_posts/0000-00-00-welcome-to-jekyll.markdown.erb"
         end
+
+        # After a new blog has been created, print a success notification and
+        # then automatically execute bundle install from within the new blog dir
+        # unless the user opts to generate a blank blog or skip 'bundle install'.
+
+        def after_install(path, options = {})
+          Jekyll.logger.info "New jekyll site installed in #{path.cyan}."
+          Jekyll.logger.info "Bundle install skipped." if options["skip_bundle"]
+
+          unless options["blank"] || options["skip_bundle"]
+            bundle_install path
+          end
+        end
+
+        def bundle_install(path)
+          Jekyll::External.require_with_graceful_fail "bundler"
+          Jekyll.logger.info "Running bundle install in #{path.cyan}..."
+          Dir.chdir(path) do
+            unless ENV["CI"] || ENV["TEST"] # skip `bundle` during local tests
+              system("bundle install")
```

Comment:
I would prefer `system("bundle", "install")` as well. You may also have to specify the `BUNDLER_GEMFILE` variable as well.